File: content/Учёба/Практика/2.1.2026/Практика/PYTHON/8.py
import sympy as sp
import numpy as np
import matplotlib.pyplot as plt

x_sym = sp.Symbol('x')
y_upper = x_sym
y_lower = x_sym**2

V_expr = sp.pi * (y_upper**2 - y_lower**2)
V_sym = sp.integrate(V_expr, (x_sym, 0, 1))

print(f"Точное значение объема: {V_sym}")
print(f"Численное значение: {float(V_sym.evalf()):.4f}")

fig = plt.figure(figsize=(10, 8))
ax = fig.add_subplot(111, projection='3d')

# Параметризация поверхностей для вращения вокруг OX
# x остается x, y_3d = R*cos(theta), z_3d = R*sin(theta)
x_val = np.linspace(0, 1, 50)
theta = np.linspace(0, 2*np.pi, 50)
X_grid, Theta_grid = np.meshgrid(x_val, theta)

# Внешний конус (от y=x)
R_out = X_grid
Y_out = R_out * np.cos(Theta_grid)
Z_out = R_out * np.sin(Theta_grid)

# Внутренний параболоид (от y=x^2)
R_in = X_grid**2
Y_in = R_in * np.cos(Theta_grid)
Z_in = R_in * np.sin(Theta_grid)

ax.plot_surface(X_grid, Y_out, Z_out, color='blue', alpha=0.3, edgecolor='none')
ax.plot_surface(X_grid, Y_in, Z_in, color='red', alpha=0.8, edgecolor='none')

ax.set_title('Тело вращения (синий - конус, красный - параболоид)')
ax.set_xlabel('Ось X')
ax.set_ylabel('Ось Y')
ax.set_zlabel('Ось Z')
ax.set_box_aspect((1, 1, 1)) 
plt.show()

File: content/Учёба/Практика/2.1.2026/Практика/MATLAB/8.m
syms x_sym;
y_upper = x_sym;      % Прямая проходит выше параболы на [0, 1]
y_lower = x_sym^2;    % Парабола 
% 1. Точки пересечения (границы интегрирования)
limits = solve(y_upper == y_lower, x_sym);
a = double(min(limits)); % 0
b = double(max(limits)); % 1
% 2. Формула объема тела вращения с отверстием внутри (шайбы):
% V = pi * int( R_upper^2 - r_lower^2 ) dx
V_expr = pi * (y_upper^2 - y_lower^2);
V_sym = int(V_expr, x_sym, a, b);
fprintf('Границы интегрирования: от x = %d до x = %d\n', a, b);
disp(['Точное значение объема: V = ', char(V_sym)]);
fprintf('Численное значение: V = %.4f\n', double(V_sym));
figure;
% Двумерная площадь, которая вращается
subplot(1, 2, 1);
x_num = linspace(-0.2, 1.2, 100);
plot(x_num, x_num, 'b-', 'LineWidth', 1.5); hold on;
plot(x_num, x_num.^2, 'r-', 'LineWidth', 1.5);
% Заливка области
fill([linspace(0,1,50) linspace(1,0,50)], ...
     [linspace(0,1,50) linspace(1,0,50).^2], 'g', 'FaceAlpha', 0.3);
grid on;
title('Вращаемая область (2D)');
xlabel('Ось X'); ylabel('Ось Y');
legend('y = x', 'y = x^2', 'Площадь сечения');
% Трехмерная визуализация самого тела вращения
subplot(1, 2, 2);
theta = linspace(0, 2*pi, 40);
x_3d = linspace(0, 1, 40);
[T, X_grid] = meshgrid(theta, x_3d);
% Поверхность от прямой y = x (внешний конус)
R_out = X_grid; 
Y_out = R_out .* cos(T);
Z_out = R_out .* sin(T);
surf(X_grid, Y_out, Z_out, 'FaceColor', 'b', 'FaceAlpha', 0.5, 'EdgeColor', 'none'); hold on;
% Поверхность от параболы y = x^2 (внутренняя воронка)
R_in = X_grid.^2; 
Y_in = R_in .* cos(T);
Z_in = R_in .* sin(T);
surf(X_grid, Y_in, Z_in, 'FaceColor', 'r', 'FaceAlpha', 0.8, 'EdgeColor', 'none');
view(30, 30);
camlight; lighting gouraud;
title('Полученное тело вращения (3D)');
xlabel('Ось X'); ylabel('Ось Y'); zlabel('Ось Z');
axis equal;